@here/harp-map-controls/lib/CameraKeyTrackAnimation.ts

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { GeoCoordinates, GeoCoordLike } from "@here/harp-geoutils";
+import { LookAtParams, MapView, MapViewEventNames, MapViewUtils } from "@here/harp-mapview";
+import { LoggerManager } from "@here/harp-utils";
+import THREE = require("three");
+
+const logger = LoggerManager.instance.create("CameraKeyTrackAnimation");
+
+export interface ControlPointOptions extends Partial<LookAtParams> {
+    /**
+     * @override
+     */
+    target: GeoCoordinates | GeoCoordLike;
+    /**
+     * A timestamp for the [[ControlPoint]], describes when during the animation the
+     * [[ControlPoint]] should be reached, in seconds
+     */
+    timestamp: number;
+    /**
+     * @override
+     */
+    tilt?: number;
+    /**
+     * @override
+     */
+    heading?: number;
+    /**
+     * @override
+     */
+    distance?: number;
+    /**
+     * A name for the ControlPoint
+     */
+    name?: string;
+}
+/**
+ * A ControlPoint used to defined the Track of the [[CameraKeyTrackAnimation]]
+ */
+export class ControlPoint
+    implements Pick<LookAtParams, "target" | "tilt" | "heading" | "distance"> {
+    /**
+     * The time, when this CameraPose should be reached in the animation flow, in seconds.
+     */
+    timestamp: number = 0;
+    target: GeoCoordinates;
+    name?: string;
+    tilt: number;
+    heading: number;
+    distance: number;
+
+    constructor(options: ControlPointOptions) {
+        this.timestamp = options.timestamp;
+        this.target = options.target
+            ? GeoCoordinates.fromObject(options.target)
+            : new GeoCoordinates(0, 0);
+        this.tilt = options.tilt ?? 0;
+        this.heading = options.heading ?? 0;
+        this.distance = options.distance ?? 0;
+        this.name = options.name ?? Date.now().toString();
+    }
+}
+
+export interface CameraKeyTrackAnimationOptions {
+    /**
+     * The [[ControlPoint]]s the animation should pass
+     */
+    controlPoints: ControlPoint[];
+    /**
+     * If ´true` the animation restarts after finishing, therefore repeats. @default false.
+     */
+    repeat?: boolean;
```

Comment:
Isn't this already doable with `loop`?

Reply:
good catch, yes, removed that one, its not even used